### PR TITLE
[Boost tier] - Create access codes

### PR DESCRIPTION
**Pull request by tudorraul**
As a singed in Patron or Partner, that opted for the events benefit, 
when I choose to book a Guardian live event eligible for free tickets,
I can also select 1 free ticket and up to 2 discounted tickets on Eventbrite frontend.

### CODE REVIEW

**Comment by tudorraul** [comment]
@rtyley

File: frontend/app/services/zuora/ZuoraRestService.scala [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    val backendEnv = config.envName
+    val service = "ZuoraRestClient"
+
+    def recordError() {
+      put("error-count", 1)
+    }
+  }
+
+  private def get(uri: String) = client.execute(new Builder()
+    .addHeader("apiAccessKeyId", config.username)
```
Comment (rtyley) [inline frontend/app/services/zuora/ZuoraRestService.scala]: Are the "apiAccessKeyId" and "apiSecretAccessKey" really the username and password - not any special token?!

https://knowledgecenter.zuora.com/BC_Developers/REST_API/A_REST_basics#Authentication

Comment (tudorraul) [inline frontend/app/services/zuora/ZuoraRestService.scala]: Simple as that, yes.

**Comment by dominickendrick** [comment]
Can we put down the test criteria in the pull request notes ? Just so we are sure what needs to be checked

File: frontend/app/services/MemberService.scala [hunk 2]
```diff
@@ -122,14 +123,28 @@ trait MemberService extends LazyLogging with ActivityTracking {
     }
   }
 
-  def createDiscountForMember(member: Member, event: RichEvent): Future[Option[EBCode]] = (for {
+  def retrieveComplimentaryTickets(member: Member, event: RichEvent): Future[Seq[EBTicketClass]] =
```
Comment (dominickendrick) [inline frontend/app/services/MemberService.scala]: Can we add curly braces for readability `Future[Seq[EBTicketClass]] = {`

File: frontend/app/services/MemberService.scala [hunk 3]
```diff
@@ -122,14 +123,28 @@ trait MemberService extends LazyLogging with ActivityTracking {
...
+    for {
+      memberTierFeatures <- TouchpointBackend.forUser(IdMinimalUser(member.identityId,None))
+        .subscriptionService.memberTierFeatures(member)
+    } yield {
+      val memberWithEventsFeature = memberTierFeatures.map(_.code).contains(FreeEventTickets.zuoraCode)
+      event.internalTicketing.map(_.complimentaryTickets).filter(ticket => memberWithEventsFeature).getOrElse(Nil)
+    }
+
+  def retrieveDiscountedTickets(member: Member, event: RichEvent): Seq[EBTicketClass] =
```
Comment (dominickendrick) [inline frontend/app/services/MemberService.scala]: Can we add curly braces for readability `Seq[EBTicketClass] = {`

**Comment by dominickendrick** [comment]
:+1: apart from above

**Pull request merged**